PyStemmusScope/stemmus_scope.py

```diff
@@ -122,37 +124,47 @@ def setup(
         Args:
             WorkDir: path to a directory where input/output directories should be created.
             ForcingFileName: forcing file name. Forcing file should be in netcdf format.
-            NumberOfTimeSteps: total number of time steps in which model runs. It can be
-                `NA` or a number. Example `10` runs the model for 10 time steps.
+            StartTime: Start time of the model run. It must be in ISO format (e.g. 2007-01-01T00:00).
+            EndTime: End time of the model run. It must be in ISO format (e.g. 2007-01-01T00:00).
 
         Returns:
-            Paths to config file and input/output directories
+            Path to the config file
+            Forcing filename
         """
         # update config template if needed
         if WorkDir:
             self._config["WorkDir"] = WorkDir
 
-        if ForcingFileName:
-            self._config["ForcingFileName"] = ForcingFileName
+        if Location:
+            _ = utils.check_location_fmt(Location)
+            self._config["Location"] = Location
 
-        if NumberOfTimeSteps:
-            self._config["NumberOfTimeSteps"] = NumberOfTimeSteps
+        if StartTime:
+            self._config["StartTime"] = StartTime
+
+        if EndTime:
+            self._config["EndTime"] = EndTime
+
+        utils.check_time_fmt(self._config)
```

Comment:
let's move this check to `config_io.read_config()` function.

Reply:
I removed this check, but I've added the validate_config check here. T did this so the config is validated *before* all the directories are created.